eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -869,14 +869,14 @@ object Router {
     val currentBlockHeight = Globals.blockCount.get()
 
     val boundaries: RichWeight => Boolean = { weight =>
-      ((weight.cost - amountMsat) < routeParams.maxFeeBaseMsat || (weight.cost - amountMsat) < (routeParams.maxFeePct * amountMsat)) &&
+      ((weight.cost - amount) < routeParams.maxFeeBase || (weight.cost - amount) < MilliSatoshi(amount.toLong * routeParams.maxFeePct.toLong)) &&
```

Comment:
Do we still need to `.toLong` calls here ?

Reply:
Not really but `maxFeePct` is a `Double` and can't be multiplied with satoshis as-is, however there is a cleaner way to do this:
```
((weight.cost - amount) < routeParams.maxFeeBase || (weight.cost - amount) < amount * routeParams.maxFeePct.toLong) &&
```